### ISSUE TITLE
Feature request: null SOLE_AUTHENTICATION_SERVICE in CoInitializeSecurity 

### ISSUE THREAD

**Issue by SecSamDev** [microsoft/windows-rs#1640]
### Motivation

I'm trying to build a WMI client.
Currently the rust API does not allow passing the value "-1" in the "cAuthSvc" parameter of the CoInitializeSecurity function, instead, an array is passed and rust uses the length of the array as the "cAuthSvc" parameter.

```rust
#[link(name = "windows")]
extern "system" {
    fn CoInitializeSecurity(psecdesc: *const super::super::Security::SECURITY_DESCRIPTOR, cauthsvc: i32, asauthsvc: *const SOLE_AUTHENTICATION_SERVICE, preserved1: *const ::core::ffi::c_void, dwauthnlevel: RPC_C_AUTHN_LEVEL, dwimplevel: RPC_C_IMP_LEVEL, pauthlist: *const ::core::ffi::c_void, dwcapabilities: EOLE_AUTHENTICATION_CAPABILITIES, preserved3: *const ::core::ffi::c_void) -> ::windows::core::HRESULT;
}
CoInitializeSecurity(::core::mem::transmute(psecdesc), asauthsvc.len() as _, ::core::mem::transmute(::windows::core::as_ptr_or_null(asauthsvc)), ::core::mem::transmute(preserved1), ::core::mem::transmute(dwauthnlevel), ::core::mem::transmute(dwimplevel), ::core::mem::transmute(pauthlist), ::core::mem::transmute(dwcapabilities), ::core::mem::transmute(preserved3)).ok()


```

### Rationale and alternatives

If the original "cauthsvc" parameter cannot be added, then some facility should be provided to generate the array of SOLE services.

**Comment by kennykerr** [comment]
This is currently blocked on a metadata bug: https://github.com/microsoft/win32metadata/issues/837

Duplicate of #1610
